app/models/settings/base.rb

```diff
@@ -0,0 +1,159 @@
+# This was taken from the rails-settings-cached gem and adapted to work with
+# more than one settings model by making the request cache class-specific. We
+# also removed features we don't need like cache scopes and readonly fields.
+#
+# See: https://github.com/huacnlee/rails-settings-cached
+module Settings
+  class Base < ActiveRecord::Base # rubocop:disable Rails/ApplicationRecord
+    self.abstract_class = true
+
+    class SettingNotFound < RuntimeError; end
+
+    class ProcetedKeyError < ArgumentError; end
+
+    PROTECTED_KEYS = %w[var value].freeze
+    SEPARATOR_REGEXP = /[\n,;]+/.freeze
+
+    # In rails-settings-cache this is defined in the railtie.
+    # See: https://github.com/huacnlee/rails-settings-cached/blob/main/lib/rails-settings/railtie.rb
+    after_commit :clear_cache, on: %i[create update destroy]
+
+    class << self
+      def clear_cache
+        RequestStore.delete(cache_key)
+        Rails.cache.delete(cache_key)
+      end
+
+      def field(key, default: nil, type: :string, separator: nil, validates: nil)
+        define_field(
+          key,
+          default: default,
+          type: type,
+          separator: separator,
+          validates: validates,
+        )
+      end
+
+      def get_field(key)
+        @defined_fields.detect { |field| field[:key] == key.to_s } || {}
+      end
+
+      def keys
+        @defined_fields.pluck(:key)
+      end
+
+      private
+
+      def cache_key
+        @cache_key ||= name.underscore
+      end
+
+      def define_field(key, default: nil, type: :string, separator: nil, validates: nil)
+        key = key.to_s
+
+        raise(ProcetedKeyError, "Can't use #{key} as setting key.") if key.in?(PROTECTED_KEYS)
+
+        @defined_fields ||= []
+        @defined_fields << {
+          key: key,
+          default: default,
+          type: type || :string
+        }
+
+        # Getter
+        define_singleton_method(key) do
+          result = __send__(:value_of, key)
```

Comment:
I refactored the code for retrieving the value and generating the default if necessary.